akka-http-core/src/main/scala/akka/http/impl/engine/parsing/HttpMessageParser.scala

```diff
@@ -161,9 +162,21 @@ private[http] trait HttpMessageParser[Output >: MessageOutput <: ParserOutput] {
           case _         => failMessageStart("HTTP message must not contain more than one Content-Length header")
         }
         case h: `Content-Type` => cth match {
-          case None      => parseHeaderLines(input, lineEnd, headers, headerCount + 1, ch, clh, Some(h), teh, e100c, hh)
-          case Some(`h`) => parseHeaderLines(input, lineEnd, headers, headerCount, ch, clh, cth, teh, e100c, hh)
-          case _         => failMessageStart("HTTP message must not contain more than one Content-Type header")
+          case None =>
+            parseHeaderLines(input, lineEnd, headers, headerCount + 1, ch, clh, Some(h), teh, e100c, hh)
+          case Some(`h`) =>
+            parseHeaderLines(input, lineEnd, headers, headerCount, ch, clh, cth, teh, e100c, hh)
+          case Some(`Content-Type`(ContentTypes.`NoContentType`)) =>
+            parseHeaderLines(input, lineEnd, headers += h, headerCount + 1, ch, clh, cth, teh, e100c, hh)
```

Comment:
Why is this one handled specially? Is that even a content-type that we would expect to see on the wire at all?

Reply:
Ah, I see it below. Maybe add a comment that this is how the `NoContentType` setting is handled.